chaviprom/secure_otp_views.py

```
@@ -0,0 +1,386 @@
+        
+        # Mark this challenge as used to prevent replay
+        self._mark_challenge_used(self.request)
+        
+        # Bind session to user context
+        self._bind_session_to_user(self.request)
+        
+        return super().form_valid(form)
+    
+    def _generate_challenge_id(self, request):
+        """Generate a unique challenge ID for this OTP attempt."""
+        challenge_data = f"{request.session.session_key}:{time.time()}:{random_hex()}"
```

Ruff (0.12.2)

Undefined name `random_hex`

(F821)